### PR TITLE
Add data-set config object from utils

### PR DESCRIPTION
**Pull request by einarmo**
Aligns with other extractors.

### CODE REVIEW

File: Extractor/Pushers/CDFPusher.cs [hunk 1]
```diff
@@ -1253,18 +1253,18 @@ private async Task MarkReferencesAsDeleted(IEnumerable<string> externalIds, Canc
         /// </summary>
         private async Task EnsureConfigInit(CancellationToken token)
         {
-            if (!config.DataSetId.HasValue && !string.IsNullOrEmpty(config.DataSetExternalId))
+            if (config.DataSet != null)
```
Comment (ozangoktan) [inline Extractor/Pushers/CDFPusher.cs:1256]: The old if block was checking if dataset was defined by externalid, and retrieving the numeric id if necessary. 

This one is checking the top level object, which is also checked in DataSets.GetId method. Should we rework this section a bit?

File: Extractor/Pushers/CDFPusher.cs [hunk 2]
```diff
@@ -1253,18 +1253,18 @@ private async Task MarkReferencesAsDeleted(IEnumerable<string> externalIds, Canc
...
-                    config.DataSetId = dataSet.First().Id;
+                    var id = await destination.CogniteClient.DataSets.GetId(config.DataSet, token);
+                    config.DataSet.Id = id;
                 }
                 catch (ResponseException ex)
                 {
                     log.LogError("Could not fetch data set by external id. It may not exist, or the user may lack" +
                         " sufficient access rights. Project {Project} at {Host}, id {Id}: {Message}",
-                        config.Project, config.Host, config.DataSetExternalId, ex.Message);
+                        config.Project, config.Host, config.DataSet.ExternalId, ex.Message);
```
Comment (ozangoktan) [inline Extractor/Pushers/CDFPusher.cs:1267]:
```suggestion
                        config.Project, config.Host, config.DataSet?.ExternalId, ex.Message);
```
This one might be unnecessary if this section is reworked.

File: Extractor/Pushers/CDFPusher.cs (hunk 1 above)
Comment (einarmo) [reply Extractor/Pushers/CDFPusher.cs:1256]: It's the same, the GetId method checks if ExternalId is defined.

**Review by ozangoktan — APPROVED** [review]
LGTM

**Pull request merged**